lib/content-services/src/lib/content-node-share/content-node-share.dialog.ts

```diff
@@ -266,25 +274,46 @@ export class ShareDialogComponent implements OnInit, OnDestroy {
         this.form.setValue({
             sharedUrl: `${this.baseShareUrl}${this.sharedId}`,
             time: expiryDate ? moment(expiryDate).local() : null
-        });
+        }, { emitEvent: false });
 
-        if (expiryDate) {
-            this.time.enable();
-        } else {
-            this.time.disable();
-        }
+        return expiryDate;
     }
 
-    private updateNode(date: moment.Moment): Observable<Node> {
+    private updateNode(date: moment.Moment) {
+
         const expiryDate = date
-            ? (this.type === 'date' ? date.endOf('day').utc().format() : date.utc().format())
+            ? (this.type === 'date' ? date.endOf('day').utc().format('YYYY-MM-DDTHH:mm:ss.SSSZ')
+            : date.utc().format('YYYY-MM-DDTHH:mm:ss.SSSZ'))
             : null;
 
-        return this.nodesApiService.updateNode(this.data.node.entry.id, {
-            properties: {
-                'qshare:expiryDate': expiryDate
-            }
-        });
+        if (this.sharedId && expiryDate) {
+                this.isDisabled = true;
+
+                this.sharedLinksApiService.deleteSharedLink(this.sharedId).subscribe((response: any) => {
+                    if (response instanceof Error) {
+                        this.isDisabled = false;
+                        this.isFileShared = true;
+                        this.handleError(response);
+                    } else {
+                        this.sharedLinkWithExpirySettings(expiryDate);
+                        this.isLinkWithExpiryDate = true;
+                        this.updateEntryExpiryDate(date);
+                    }
+            });
+        }
+    }
+
+    private sharedLinkWithExpirySettings(expiryDate: Date | string) {
+        if (typeof expiryDate === 'string') {
+            const lastIndex = expiryDate?.lastIndexOf(':');
+            expiryDate = expiryDate?.substring(0, lastIndex) + expiryDate?.substring(lastIndex + 1, expiryDate?.length);
+        }
+        const nodeObject = {
```

Comment:
Could you please set type of this object? I think type should be SharedLinkBodyCreate

Reply:
specified